apps/rush-lib/src/api/PackageJsonEditor.ts

```diff
@@ -9,6 +9,18 @@ import {
   Sort
 } from '@microsoft/node-core-library';
 
+/**
+ * @beta
+ */
+export interface IDependencyFileEditor {
```

Comment:
This is not a good interface contract.  Too much code is required to implement this contract correctly, and the contract itself implies a specific implementation.  Maybe an abstract base class would be better for sharing code?

Reply:
Hmmm...  actually, on closer inspection it seems that this abstraction is fairly specific to the `VersionMismatchFinder` algorithm, which already has wrapper objects (`IVersionMismatchFinderEntity`).  I'm not sure the commonality between **package.json** and **common-versions.json** are significant enough to be part of the public API for **rush-lib**.

I'll follow up offline.